web/packages/teleterm/src/services/tshd/types.ts

```diff
@@ -59,6 +59,27 @@ export interface Server extends apiServer.Server.AsObject {
 
 export interface App extends apiApp.App.AsObject {
   uri: uri.AppUri;
+  /** Name of the application. */
+  name: string;
+  /** URI and port the application is available at. */
+  endpointUri: string;
```

Comment:
So does `endpointUri` always point to a subdomain within the cluster while `publicAddr` can potentially be an address available to the internet? This also would be worthwhile to capture in comments.

Reply:
Okay, so I was a little wrong.
`endpointUri` is a "physical" address of the app, like  "http://127.0.0.1:54344".
Then the public address by default is a subdomain, like "dumper.proxy.127.0.0.1.nip.io". 
You can override this in the config file, by setting `public_addr` to anything you want.

But I'm not sure when a `publicAddr` can be empty (except of SAML apps). @kimlisa do you know this? I based on this comment https://github.com/gravitational/teleport/blob/b178b8b7322e7c6348bbd17ed1d3251ad34d946b/web/packages/teleport/src/services/apps/types.ts#L35-L36

Reply:
Okay, [looks like `publicAddr` is non-empty](https://github.com/gravitational/teleport/blob/5a94e3aa59727a1e5ffeccac8d8224ada3924882/lib/srv/app/watcher.go#L133) only for SAML apps. So the updated comment is correct.